discussion_forum/tests.py

```diff
@@ -0,0 +1,592 @@
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+
+from .models import DiscussionComment, DiscussionReply, DiscussionThread
+
+
+ins_cred = {"email": "instructor@bodhitree.com", "password": "instructor"}
+
+
+class DiscussionThreadViewSetTest(APITestCase):
+    """Test for DiscussionThreadViewSet."""
+
+    fixtures = [
+        "users.test.yaml",
+        "courses.test.yaml",
+        "coursehistories.test.yaml",
+        "discussionforum.tests.yaml",
+        "discussionthread.tests.yaml",
+    ]
+
+    def _list_discussion_threads_helper(self):
+        """helper function to test list discussion threads functionality."""
+
+        url = reverse(
+            "discussion_forum:discussionthread-list-discussion-threads", args=[1]
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        length = DiscussionThread.objects.all().count()
+        self.assertEqual(len(response.data), length)
+
+    def test_list_discussion_threads(self):
+        """
+        Test to check: list all discussion_threads.
+        """
+        self.client.login(**ins_cred)
+        self._list_discussion_threads_helper()
+        self.client.logout()
+
+    def _retrieve_discussion_thread_helper(self, discussion_thread_id):
+        """helper function to test retrive discussion thread functionality."""
+
+        url = reverse(
+            "discussion_forum:discussionthread-retrieve-discussion-thread",
+            kwargs={"pk": discussion_thread_id},
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["id"], discussion_thread_id)
+
+    def test_retrieve_discussion_thread(self):
+        """
+        Test to check: retrieve discussion_thread.
+        """
+        discussion_thread_id = 1
```

Comment:
why pass as a parameter when its same for the call

Reply:
discussion_thread_id, define it local var in called method

Reply:
done